java8/src/main/code_generator/I18n.java8.txt

```diff
@@ -1,28 +1,37 @@
 package cucumber.api.java8;
 
-import cucumber.runtime.java8.ConstantPoolTypeIntrospector;
-import cucumber.runtime.java8.LambdaGlueBase;
 import cucumber.runtime.java.JavaBackend;
+import cucumber.runtime.java8.Java8StepDefinition;
+import cucumber.runtime.java8.LambdaGlueBase;
 
 public interface ${className} extends LambdaGlueBase {
 <% i18n.stepKeywords.findAll { !it.contains('*') && !it.matches("^\\d.*") }.sort().unique().each { kw -> %>
     default void ${java.text.Normalizer.normalize(kw.replaceAll("[\\s',!]", ""), java.text.Normalizer.Form.NFC)}(final String regexp, final StepdefBody.A0 body) {
-        JavaBackend.INSTANCE.get().addStepDefinition(regexp, 0, body, ConstantPoolTypeIntrospector.INSTANCE);
+        JavaBackend.INSTANCE.get().addStepDefinition(
+            new Java8StepDefinition(regexp, 0, StepdefBody.A0.class, body)
+        );
     }
 
     default void ${java.text.Normalizer.normalize(kw.replaceAll("[\\s',!]", ""), java.text.Normalizer.Form.NFC)}(final String regexp, final long timeoutMillis, final StepdefBody.A0 body) {
-        JavaBackend.INSTANCE.get().addStepDefinition(regexp, timeoutMillis, body, ConstantPoolTypeIntrospector.INSTANCE);
+        JavaBackend.INSTANCE.get().addStepDefinition(
+            new Java8StepDefinition(regexp, timeoutMillis, StepdefBody.A0.class, body)
```

Comment:
Typetools resolves the arguments of the the generic type (StepdefBody.A1) using the type variable information from body (the implementing class) so we pass these in here.